network-api/networkapi/wagtailpages/pagemodels/products.py

```diff
@@ -533,9 +533,35 @@ def categories_page(self, request, slug):
         })
 
     def get_sitemap_urls(self, request):
-        # TODO add category URLs
-        urls = super().get_sitemap_urls(request)
-        return urls
+        """
+        Add categories and route views to the sitemap for better search indexing.
+        """
+        sitemap = super().get_sitemap_urls(request)
+        last_modified = self.last_published_at or self.latest_revision_created_at
+        # Add all the available Buyers Guide categories to the sitemap
+        categories = BuyersGuideProductCategory.objects.filter(hidden=False)
+        for category in categories:
+            sitemap.append(
+                {
+                    "location": self.full_url + self.reverse_subpage("category-view", args=(category.slug,)),
+                    "lastmod": last_modified,
+                }
+            )
+
+        # Add all the available "subpages" (routes) to the sitemap, excluding categories and products.
+        # Categories are added above. Products will be their own Wagtail pages and get their own URLs because of that.
```

Comment:
This sounds like there's no plan to uplift the categories and homepage... which would be odd. Did we not plan for those?

Reply:
There's just no need to uplift the categories. They're already an existing Django model, and if we "convert" them they'd still be plain Django model but they'd have a `@register_snippet` decorator. Unless we wanted to add more to them or do some larger migration work, like moving which app the category model lives in (which would change the database table name). Thoughts?

Reply:
If we're uplifting to a wagtail app, I don't think it makes sense to uplift the homepage, and the product pages, but not the category page: categories are effectively subclasses of the homepage, with the same template but with an overwrite in the context to make sure it uses a reduced product set. So to me that feels like "these are pages, and children of the buyersguide homepage", with product pages pointing to these new category pages.

So not "move the category model" but "make a new page type for categories, and use that, instead"?